# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/TextDirectiveCreator.cpp
# Commit: 87e0ba646802
# Full Hash: 87e0ba646802000c24e32db49375d7e5dbfb2f59
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-07-31 04:35:31
# Description:
#   Bug 1979474 - Text Fragments: Normalize range boundaries when extending to word boundaries in the punctuation-only case. r=dom-core,mccr8
#   
#   Bug 1976480 introduced a special code path if the target range only consists of punctuation.
#   However, this new code path did not "normalize" the boundary points, ie. moving them to
#   the actual text node. Therefore, the target range could contain empty blocks
# ==============================================================================

diff -r 0247ec049da5 -r 87e0ba646802 dom/base/TextDirectiveCreator.cpp
--- a/dom/base/TextDirectiveCreator.cpp	Wed Jul 30 14:15:59 2025 +0000
+++ b/dom/base/TextDirectiveCreator.cpp	Wed Jul 30 14:18:26 2025 +0000
@@ -118,8 +118,11 @@
   }
   if (std::all_of(rangeContent.View().cbegin(), rangeContent.View().cend(),
                   IsPunctuationForWordSelect)) {
-    RefPtr range =
-        StaticRange::Create(aRange->StartRef(), aRange->EndRef(), rv);
+    RangeBoundary startPoint = TextDirectiveUtil::FindNextNonWhitespacePosition<
+        TextScanDirection::Right>(aRange->StartRef());
+    RangeBoundary endPoint = TextDirectiveUtil::FindNextNonWhitespacePosition<
+        TextScanDirection::Left>(aRange->EndRef());
+    RefPtr range = StaticRange::Create(startPoint, endPoint, rv);
     if (MOZ_UNLIKELY(rv.Failed())) {
       return Err(std::move(rv));
     }